broker-core/src/test/java/org/apache/qpid/server/security/encryption/AESGCMKeyFileEncrypterTest.java

```diff
@@ -0,0 +1,364 @@
+        {
+            _workDir = Files.createTempDirectory("qpid_work_dir");
+            _configurationLocation = Files.createTempFile(_workDir, "config", ".json");
+            Map<String, Object> context = new HashMap<>();
+            context.put("qpid.work_dir", _workDir.toFile().getAbsolutePath());
+            Map<String, Object> config = new HashMap<>();
+            config.put(ConfiguredObject.NAME, getTestName());
+            config.put(Broker.MODEL_VERSION, BrokerModel.MODEL_VERSION);
+            config.put(Broker.CONFIDENTIAL_CONFIGURATION_ENCRYPTION_PROVIDER, encrptionType);
+            new ObjectMapper().writeValue(_configurationLocation.toFile(), config);
+
+            attributes.put("storePath", _configurationLocation.toFile().getAbsolutePath());
+            attributes.put("preferenceStoreAttributes", "{\"type\": \"Noop\"}");
+            attributes.put(SystemConfig.TYPE, JsonSystemConfigImpl.SYSTEM_CONFIG_TYPE);
+            attributes.put(SystemConfig.STARTUP_LOGGED_TO_SYSTEM_OUT, Boolean.FALSE);
+            attributes.put(SystemConfig.CONTEXT, context);
+            final SettableFuture<SystemConfig<?>> configFuture = SettableFuture.create();
+            _systemLauncher = new SystemLauncher(new SystemLauncherListener.DefaultSystemLauncherListener()
+            {
+                @Override
+                public void onContainerResolve(final SystemConfig<?> systemConfig)
+                {
+                    configFuture.set(systemConfig);
+                }
+            });
+            _systemLauncher.startup(attributes);
+            final SystemConfig<?> systemConfig = configFuture.get(10, TimeUnit.SECONDS);
+            _broker = (Broker<?>) systemConfig.getContainer();
+
+            final Map<String, Object> authProviderAttributes = new HashMap<>();
+            authProviderAttributes.put(ConfiguredObject.NAME, "testAuthProvider");
+            authProviderAttributes.put(ConfiguredObject.TYPE, "Plain");
+
+            final AuthenticationProvider
+                    authProvider = _broker.createChild(AuthenticationProvider.class, authProviderAttributes);
+
+            Map<String, Object> userAttrs = new HashMap<>();
+            userAttrs.put(User.TYPE, "managed");
+            userAttrs.put(User.NAME, "guest");
+            userAttrs.put(User.PASSWORD, PLAINTEXT);
+            User user = (User) authProvider.createChild(User.class, userAttrs);
+        }finally{
+            _systemLauncher.shutdown();
+            FileUtils.deleteDirectory(_workDir.toFile().getAbsolutePath());
+            FileUtils.deleteDirectory(_configurationLocation.toFile().getAbsolutePath());
+        }
+    }
+
+    @Test
+    public void testRepeatedEncryptionsReturnDifferentValues() throws Exception
+    {
+        SecretKeySpec secretKey = createSecretKey();
+        AESGCMKeyFileEncrypter encrypter = new AESGCMKeyFileEncrypter(secretKey);
+
+        Set<String> encryptions = new HashSet<>();
+
+        int iterations = 100;
```

Comment:
why 100 iterations are actually needed here? what that solves?

Reply:
It cna be reduced to 10 too.. Just wanted to prove that everytime the encrypted string is different